pytorch_lightning/utilities/device_parser.py

```diff
@@ -184,11 +184,13 @@ def _normalize_parse_gpu_input_to_list(gpus: Union[int, List[int], Tuple[int, ..
     return list(range(gpus))
 
 
-def _get_all_available_gpus() -> List[int]:
+def _get_all_available_gpus() -> Optional[List[int]]:
     """
     Returns:
          a list of all available gpus
     """
+    if not torch.cuda.is_available():
+        return None
```

Comment:
should this return instead

```suggestion
        return []
```

Reply:
that is what it does now, so maybe we shall fix it elsewhere?
```py
list(range(0)) # -> []
```

Reply:
Yeah, that would work too. Worth investigating